cwms-data-api/src/main/java/cwms/cda/data/dao/ClobDao.java

```diff
@@ -107,7 +106,7 @@ public Clobs getClobs(String cursor, int pageSize, String officeLike,
                             .from(v_clob)
                             .join(v_office).on(v_clob.OFFICE_CODE.eq(v_office.OFFICE_CODE))
                             .where(JooqDao.caseInsensitiveLikeRegex(v_clob.ID, idRegex))
-                            .and(officeLike == null ? noCondition() : DSL.upper(v_office.OFFICE_ID).like(officeLike.toUpperCase()));
+                            .and(officeLike == null ? noCondition() : v_office.OFFICE_ID.like(officeLike.toUpperCase()));
```

Comment:
While the column itself is always upper case, the Upper on the column is a hint to the query planner. The indexes are generally all `create index... upper(column)` and the planner doesn't always do a great job of realizing that.

If you can show an explain plan that shows this isn't the case that's a good reason to change. But other wise the hint should stay.

However a change that should be made here is that instead of String::toUpperCase the DSL.upper should also be used on the input, the planner can then decide on if it needs to do the conversion or can just do a case insensitive compare.

Though for the column it should be v_office.OFFICE_ID.upper() I think, DSL.upper has some consequences.

But the TLDR, Query planners often don't make good decisions and need a lot of hints.

Reply:
Ok.  Good point re String::toUpperCase and jOOQ upper.   Looks like Field.upper is deprecated.